### DISCUSSION TITLE
Possible bug or I don't understand this

### DISCUSSION THREAD

**Discussion by OnyxIdol** [kagenash1/godot-behavior-tree#16] [Q&A]
Hi,
I've been building a tree with your plugin, thank you for your work! However, on execution I am getting an error on a BTSelector node. The child leaf returns suceed(), and this causes a problem because the BTSelector expects a GDScriptFunctionState as a result from its children:
```
var result:GDScriptFunctionState
	
	for c in children:
		bt_child = c
		
		result = bt_child.tick(agent, blackboard)
```

 This is a different behavior than other compositors which take an untyped result and only *then* check if it's a GDScriptFunctionState. So I'm assuming this is not the correct behavior, or am I missing something?

Edit: I removed the typed declaration and now it works as expected.

Cheers, Onyx

**Comment by kagenash1** [comment]
I'm not sure I understand what's on your end, but in general btnodes return either a function state if they are running or a bool (succeed() and fail()) if not running (hence completed). Are you defining your own custom composite in the code you showed above?  
Anyway, in general avoid static typing the return variable where you store the result of the tick() call, because that can be either a GDScriptFunctionState or a bool, and the tree handles the two cases autonomously.

Cheers, Kage

**Reply by OnyxIdol** [reply]
I didn't define a custom composite. The error exists in the predefined BTSelector, line 8.

**Reply by kagenash1** [reply]
I don't see the line you are referring to in line 8, are you sure you are using the latest stable release?

**Reply by OnyxIdol** [reply]
I am using the Version in the AssetLib, is that the latest?

**Reply by kagenash1** [reply]
The version in the asset lib is aligned to the latest release. Maybe you downloaded it long ago?

**Reply by OnyxIdol** [reply]
Nope, just yesterday or so. I'm not sure where this difference comes from, but seeing as I got it to work we probably don't need to invest more time to investigate this ;) Perhaps I can figure it out tomorrow after sleeping, but it isn't an issue anymore right now.

**Reply by kagenash1** [reply]
I was just worrying about the asset lib being messed up. Maybe clone again for safety, but if it works it works :)
